test/wanda_web/controllers/catalog_controller_test.exs

```diff
@@ -14,5 +14,15 @@ defmodule WandaWeb.CatalogControllerTest do
       api_spec = ApiSpec.spec()
       assert_schema(json, "CatalogResponse", api_spec)
     end
+
+    test "listing the checks catalog produces a CatalogResponse when filtered", %{conn: conn} do
+      json =
+        conn
+        |> get("/api/checks/catalog?provider=azure&foo=bar")
+        |> json_response(200)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "CatalogResponse", api_spec)
```

Comment:
we should test at least the length of the response here, to see if it matches the expected number of checks